src/main/java/com/careerfair/q/workflow/database/implementation/EmployeeFirebaseWorkflowImpl.java

```diff
@@ -85,14 +86,50 @@ public void registerStudentToEmployee(String employeeId, String studentId)
         Firestore firestore = FirestoreClient.getFirestore();
         Employee employee = getEmployeeWithId(employeeId);
 
-        if (employee.getStudents() == null) {
-            employee.setStudents(Lists.newArrayList());
+        if (employee.students == null) {
+            employee.students = Lists.newArrayList();
         }
-
-        if (!employee.getStudents().contains(studentId)) {
-            employee.getStudents().add(studentId);
+        if(!employee.students.contains(studentId)) {
+            employee.students.add(studentId);
             firestore.collection(EMPLOYEE_COLLECTION).document(employeeId).update("students",
-                    employee.getStudents());
+                    employee.students);
         }
     }
-}
+
+    @Override
+    public Employee updateEmployee(String employeeId, Employee updatedEmployee)
+            throws FirebaseException {
+        Firestore firestore = FirestoreClient.getFirestore();
+
+        try {
+            firestore.collection(EMPLOYEE_COLLECTION).document(employeeId)
+                    .set(updatedEmployee, SetOptions.mergeFields(UPDATE_FIELDS)).get();
+            return getEmployeeWithId(employeeId);
+        } catch (ExecutionException | InterruptedException ex) {
+            throw new FirebaseException(ex.getMessage());
+        }
+    }
+
+    @Override
+    public Employee addEmployee(Employee newEmployee) throws FirebaseException {
+        Firestore firestore = FirestoreClient.getFirestore();
+        DocumentReference documentReference = firestore.collection(EMPLOYEE_COLLECTION).document();
+        String employeeId = documentReference.getId();
+
+        // TODO: this currently will also add a field students and set that to null
```

Comment:
There is  a TODO here

Reply:
yes leaving this for now

Reply:
for reference later if issues come while testing end to end

Reply:
It doesn't seem like a big task. Is it possible to do it in this PR?

Reply:
yes- do you want to not include students as a null field when adding a new employee? Or is it fine if students is set to null (both cases would work technically)